src/script/components/MessagesList/Message/ContentMessage/asset/TextMessageRenderer.tsx

```diff
@@ -76,90 +70,64 @@ export const TextMessageRenderer: FC<TextMessageRendererProps> = ({
-        };
-
-        return event.type === 'keydown'
-          ? handleKeyEvent(event as KeyboardEvent, 'mention', mentionMsgDetails)
-          : onMessageClick(asset, event as MouseEvent, 'mention', mentionMsgDetails);
-      }
-    };
-
-    function addEventListener(elements: Element[]) {
-      elements?.forEach(element => {
-        events.forEach(eventName => {
-          element.addEventListener(eventName, handleMsgEvent);
-        });
+  const forwardEvent = (
+    event: KeyboardEvent | MouseEvent,
+    elementType: ElementType,
+    messageDetails: MessageDetails,
+  ) => {
+    if (event.type === 'keydown' && isCurrentConversationFocused) {
+      handleKeyDown(event as KeyboardEvent, () => {
+        event.preventDefault();
+        onMessageClick(event, elementType, messageDetails);
       });
+    } else if (event.type === 'click') {
+      event.preventDefault();
+      onMessageClick(event, elementType, messageDetails);
     }
+  };
 
-    function removeEventListener(elements: Element[]) {
-      elements?.forEach(element => {
-        events.forEach(eventName => {
-          element.removeEventListener(eventName, handleMsgEvent);
-        });
-      });
+  const handleClick = (event: React.MouseEvent | React.KeyboardEvent) => {
+    const target = event.target as HTMLElement;
+    if (!target) {
+      return;
     }
-
-    addEventListener(emailLinks);
-    addEventListener(markdownLinkTargets);
-    addEventListener(msgMentions);
-
-    return () => {
-      removeEventListener(emailLinks);
-      removeEventListener(markdownLinkTargets);
-      removeEventListener(msgMentions);
-    };
-  }, [onMessageClick, asset, isCurrentConversationFocused, containerRef]);
+    const isEmail = target.closest('[data-email-link]');
+    const isMarkdownLink = target.closest('[data-md-link]');
+    const isMention = target.closest('.message-mention');
+
+    if (isEmail || isMarkdownLink) {
+      const href = (event.target as HTMLAnchorElement).href;
+      const markdownLinkDetails = {
+        href: href,
+      };
+      forwardEvent(event.nativeEvent, isEmail ? 'email' : 'markdownLink', markdownLinkDetails);
+    } else if (isMention) {
+      const mentionMsgDetails = {
+        userId: target.dataset.userId,
+        userDomain: target.dataset.userDomain,
+      };
+      forwardEvent(event.nativeEvent, 'mention', mentionMsgDetails);
+    }
+  };
 
   return (
+    // We will register the click event on the paragraph element and determine the type of the element clicked.
+    // This is because the paragraph element is fed with raw HTML and we cannot register the click event on the clickabled elements directly.
+    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
     <p
       ref={setContainerRef}
       key={`${editedTimestamp}:${text}`}
+      onClick={handleClick}
+      onAuxClick={handleClick}
+      onKeyDown={handleClick}
+      onKeyUp={handleClick}
```

Comment:
I think we should come up with some better naming for `handleClick` since it handles more events that just `click`. What about `handleInteraction` or sth like that? 😄